test/browser/spec/ol/interaction/snap.test.js

```diff
@@ -533,15 +533,24 @@ describe('ol.interaction.Snap', function () {
       };
 
       const event = {
-        pixel: [30, 30],
+        pixel: [30 + width / 2, height / 2 - 30],
         coordinate: [30, 30],
         map: map,
       };
 
+      let doneCnt = 0;
       snapInteraction.on('unsnap', function (snapEvent) {
         expect(snapEvent.feature).to.be(point1);
         expect(snapEvent.segment).to.be(null);
-        done();
+        doneCnt += 1;
```

Comment:
Instead of introducing a `doneCnt` variable with a count, it would be better to ensure the correct unsnap - snap sequence, e.g. with something like
```js
const sequence = [];
function updateSequence(event) {
  sequence.push(event);
  if (sequence.length === 2) {
    expect(sequence).to.eql(['unsnap', 'snap']);
    done();
  }
}
```
and call that function in each of the listeners:
```js
updateSequence(snapEvent.type);
```

Reply:
Thanks for the review and suggestion. I checked the code and read that `unsnap` is fired before `snap` but wasn't sure if that sequence is in our specification. I've updated the code.